test/functional/wallet_backwards_compatibility.py

```diff
@@ -128,6 +128,77 @@ def test_v19_addmultisigaddress(self):
         wallet = node_v19.get_wallet_rpc("w1_v19")
         assert wallet.getaddressinfo(address_18075)["solvable"]
 
+
+    def test_v22_inactivehdchain_path(self):
+        if self.options.descriptors:
+            return
+
+        self.log.info("Testing inactive hd chain bad derivation path cleanup")
+        # 0.21.x and 22.x would both produce bad derivation paths when topping up an inactive hd chain
+        # Make sure that this is being detected and automatically cleaned up
+        node_master = self.nodes[1]
+        node_v22 = self.nodes[self.num_nodes - 6]
+        wallet_name = "bad_deriv_path"
+        node_v22.createwallet(wallet_name=wallet_name, descriptors=False)
+        wallet = node_v22.get_wallet_rpc(wallet_name)
+
+        # Make a dump of the wallet to get an unused address
+        dump_path = node_v22.wallets_path / f"{wallet_name}.dump"
+        wallet.dumpwallet(dump_path)
+        addr = None
+        seed = None
+        with open(dump_path, encoding="utf8") as f:
+            for line in f:
+                if "hdkeypath=m/0'/0'/9'" in line:
```

Comment:
I spent some time trying to understand why this one is picked here only to realise later that 9 is the last index at this chain because of keypool size set to 10 at the top of this test here: https://github.com/bitcoin/bitcoin/pull/29124/commits/e92dc1a3d36be02a8b16eed26159a43cfbe3dbcc#diff-b5055a65809b46e2775f6e28aa6eb8171e3fcc7a535cce5687a548b9ff233669R44

This 9 and 10 can be tied to each other with a constant for better readability.

Reply:
Added a constant.